processor/marshalprocessor/factory.go

```diff
@@ -0,0 +1,62 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package marshalprocessor
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/processor"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+)
+
+// componentType is the value of the "type" key in configuration.
+var componentType = component.MustNewType("marshal")
+
+const (
+	stability = component.StabilityLevelAlpha
+)
```

Comment:
I don't think we've done this for one of our processors yet, but let's start using mdatagen to generate these.

e.g. random example from an upstream processor: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/intervalprocessor/metadata.yaml

Reply:
Sounds good, took a pass on this